src/event_timer.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""EventTimer for scheduling dispatch of juju event on regular intervals."""
+
+import subprocess  # nosec B404
+from pathlib import Path
+from typing import Optional, TypedDict
+
+from jinja2 import Environment, FileSystemLoader
+
+
+class TimerEnableError(Exception):
+    """Raised when unable to enable a event timer."""
+
+
+class TimerDisableError(Exception):
+    """Raised when unable to disable a event timer."""
```

Comment:
How come we have these two exception classes here, but a separate `src/errors.py` file. Should we move these to there?

Reply:
Yes. Same as above.